pkg/sensors/platform/source/hmc.go

```diff
@@ -16,21 +13,127 @@ limitations under the License.
 
 package source
 
+import (
+	"os"
+	"strconv"
+ 
+	"k8s.io/klog/v2"
+	"github.com/sustainable-computing-io/kepler/pkg/sensors/components/source"
+	"github.com/zhmcclient/golang-zhmcclient/pkg/zhmcclient"
+)
+
+var hmcManager *zhmcclient.ZhmcManager
+
 type PowerHMC struct{}
 
-func (a *PowerHMC) GetName() string {
-	return "hmc"
+func (r *PowerHMC) GetName() string {
+    return "hmc"
+}
+
+func (r *PowerHMC) GetHMCManager() *zhmcclient.ZhmcManager {
+	if hmcManager == nil {
+		endpoint := os.Getenv("HMC_ENDPOINT")
```

Comment:
not sure we should add a PREFIX such as Z_ so it will be easier for other platform to know it's for z only?